### DISCUSSION TITLE
PLEASE HELP! Cannot access dashboard on AWS EC2 instance!!!!!

### DISCUSSION THREAD

**Discussion by shrihari-prakash** [chaos-mesh/chaos-mesh#4210] [Q&A]
I am trying to install Chaos Mesh on an EC2 instance (also tried on Azure VM) to perform chaos testing on another open shift cluster. I installed Chaos Mesh using this documentation: https://chaos-mesh.org/docs/production-installation-using-helm/.

After I did the installation (on both minikube and microk8s), quite literally on 6 fresh VMs, this is how the status of pods look:

```
root@ip-xxx-xxx-xxx:~# kubectl get pods -n chaos-mesh
NAME                                       READY   STATUS    RESTARTS      AGE
chaos-dns-server-779499656c-pcnk4          1/1     Running   0             46m
chaos-controller-manager-8d6c7c65c-jczkb   1/1     Running   0             46m
chaos-dashboard-77f5cf9985-ggfzv           1/1     Running   0             46m
chaos-controller-manager-8d6c7c65c-v4xjc   1/1     Running   1 (46m ago)   46m
chaos-controller-manager-8d6c7c65c-72ztz   1/1     Running   1 (46m ago)   46m
chaos-daemon-v2k66                         1/1     Running   0             46m
```

If I run `kubectl get svc -n chaos-mesh`, there is a perfectly configured NodePort with the following output:

```
NAME                            TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                 AGE
chaos-daemon                    ClusterIP   None             <none>        31767/TCP,31766/TCP                     47m
chaos-mesh-dns-server           ClusterIP   10.152.xx   <none>        53/UDP,53/TCP,9153/TCP,9288/TCP         47m
chaos-mesh-controller-manager   ClusterIP   10.152.xxx   <none>        443/TCP,10081/TCP,10082/TCP,10080/TCP   47m
chaos-dashboard                 NodePort    10.152.xxx   <none>        2333:32251/TCP,2334:32437/TCP           47m
```

On my Security Group, I have ports 2333, 32251, 2334, 32437 all allowed from everywhere. (also tried allowing all ports from all places as well).

On the dashboard logs, I see that the dashboard is listening also:

```
...
...
...
2023-09-22T13:17:16.773Z        INFO    ttlcontroller   ttlcontroller/ttlcontroller.go:63       Starting database TTL controller
2023-09-22T13:17:16.773Z        INFO    fx      fx-logr@v0.1.0/logr.go:44       started
[GIN-debug] [WARNING] You trusted all proxies, this is NOT safe. We recommend you to set a value.
Please check https://pkg.go.dev/github.com/gin-gonic/gin#readme-don-t-trust-all-proxies for details.
***** *[GIN-debug] Listening and serving HTTP on 0.0.0.0:2333 ******* <<<<<<<--------- LISTENING!!!!!!!!!!!!!!!!!!!
2023-09-22T13:17:16.773Z        INFO    collector       collector/server.go:153 Starting collector
2023-09-22T13:17:16.773Z        INFO    ttlcontroller   ttlcontroller/ttlcontroller.go:70       Deleting expired data from the database
2023-09-22T13:17:16.773Z        INFO    manager/internal.go:369 Starting server {"path": "/metrics", "kind": "metrics", "addr": "[::]:2334"}
...
...
...
```

In addition to that I also tried port forwarding using the command `kubectl port-forward -n chaos-mesh svc/chaos-dashboard 2333:2333`

Absolutely frustrating experience. :(

So then I visit the URL http://xx.xxx.xxx.xxx:2333/ (and allllllll the other ports that are exposed). But I am getting a cannot reach page error in browser. Please tell me what am I doing wrong? Any help is highly appreciated. Thanks in advance.

**Comment by pravalikaponugupati** [comment]
I am facing the same issue...

**Comment by STRRL** [comment]
Hi @shrihari-prakash @pravalikaponugupati , networking policy and connectivity are extremely complex, especially in the Cloud environment. I could not tell you how to make it work just from the information your provided.

Take are on the firewall rules, security groups rules and network policies, and maybe you could use AWS Reachability Analyzer to profile the issue.

BTW, I noticed that it also not works with `kubectl port-forward`, it does not make sense. I think this might tightly related to the root cause, but I have no idea why it could be.
